### PR TITLE
[LOGGER] Remove all log_runtime in project

### PR DESCRIPTION
**Pull request by benliao1**
Removed `log_runtime` from project and converted to `log_printf`. `log_runtime` is not in `logger` any more either.

### CODE REVIEW

File: logger/logger.c [hunk 1]
```diff
@@ -217,12 +216,27 @@ void logger_stop ()
...
 	static time_t now;                   //for holding system time
 	static char *time_str;               //for string representation of system time
 	static char final_msg[MAX_LOG_LEN];  //final message to be logged to requested locations
 	static int len;                      //holding lengths of various strings
+	static char msg[MAX_LOG_LEN - 100];  //holds the expanded format string (100 to make room for log header)
+	va_list args;                        //this holds the variable-length argument list
+	
+	//expands the format string into msg
+	va_start(args, format);
+	vsprintf(msg, format, args);
```
Comment (AVSurfer123) [inline logger/logger.c:238]: probably should put this after you check if the level is high enough

File: shm_wrapper/shm_wrapper.c [hunk 2]
```diff
@@ -278,26 +265,31 @@ void shm_init (process_t process)
 		
 		//mutual exclusion semaphore, catalog_mutex with initial value = 0
 		if ((catalog_sem = sem_open(CATALOG_MUTEX_NAME, O_CREAT, 0660, 0)) == SEM_FAILED) {
-			error("sem_open: catalog_mutex@dev_handler");
+			log_printf(FATAL, "sem_open: catalog_mutex@dev_handler");
```
Comment (AVSurfer123) [inline shm_wrapper/shm_wrapper.c:268]: For some of these, I see you don't include the `strerror(errno)` so just make sure you add those to all the system C functions.

File: logger/logger.c (hunk 1 above)
Comment (benliao1) [reply logger/logger.c:238]: Good point. Fixed

File: shm_wrapper/shm_wrapper.c (hunk 2 above)
Comment (benliao1) [reply shm_wrapper/shm_wrapper.c]: 👌

**Pull request merged**